File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.begin(),citations.end());
    for(int i=0;i<=1000;i++) {
        int tempA=0;
        for(int j=0;j<citations.size();j++) {
            if(i<=citations[j]) {
                tempA+=1;
            }
        }
        answer=max(answer,min(i,tempA));
    }
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        List<Integer> arr = new ArrayList<>();
        for(int i=0;i<citations.length;i++) {
            arr.add(citations[i]);
        }
        Collections.sort(arr);
        
        for(int i=0;i<=1000;i++) {
            int tempA=0;
            for(int j=0;j<citations.length;j++) {
                if(i<=citations[j]) {
                    tempA+=1;
                }
            }
            answer=Math.max(answer,Math.min(i,tempA));
        }
        
        return answer;
    }
}